File: api/src/main/java/net/blueshell/api/dto/TelemetryDTO.java
package net.blueshell.api.dto;


import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;
import lombok.EqualsAndHashCode;
import net.blueshell.api.base.BaseDTO;
import net.blueshell.api.common.enums.PlatformType;

import java.sql.Timestamp;
import java.time.Instant;

@Data
@EqualsAndHashCode(callSuper = true)
@Schema(name = "Telemetry")
public class TelemetryDTO extends BaseDTO {
    private String id;
    private String url;
    private PlatformType platform;
    private Instant createdAt;
}

File: api/src/main/kotlin/net/blueshell/api/dto/TelemetryDTO.kt
package net.blueshell.api.dto

import io.swagger.v3.oas.annotations.media.Schema
import net.blueshell.api.common.enums.PlatformType
import net.blueshell.api.dto.base.AuditedAutoIdDTO

@Schema(name = "Telemetry")
data class TelemetryDTO(
    var url: String? = null,
    var platform: PlatformType? = null
) : AuditedAutoIdDTO()
